components/switch/tests/unit/switch.spec.tsx

```diff
@@ -0,0 +1,100 @@
+import { GEL } from '@westpac/core';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Switch } from '@westpac/switch';
+import wbc from '@westpac/wbc';
+import { overridesTest } from '../../../../helpers/tests/overrides-test';
+import { nestingTest } from '../../../../helpers/tests/nesting-test';
+import { SwitchProps } from '../../src/Switch';
+
+overridesTest({
+	name: 'switch',
+	overrides: ['Switch', 'Label', 'Toggle'],
+	Component: ({ label, ...rest }: SwitchProps) => <Switch label="A Switch" {...rest}></Switch>,
+});
+
+nestingTest({
+	name: 'switch',
+	Component: ({ label, ...rest }: SwitchProps) => <Switch label="A Switch" {...rest}></Switch>,
+});
+
+// Component specific tests
+describe('Switch component', () => {
+	const SimpleSwitch = (props: SwitchProps) => (
+		<GEL brand={wbc}>
+			<Switch {...props} data-testid="switch-parent" />
+		</GEL>
+	);
+
+	test('It should render Switch', () => {
+		const { container } = render(<SimpleSwitch label="A Switch" />);
+		expect(container).toBeInTheDocument();
+	});
+
+	test('should change its checked valued when clicked', async () => {
+		const handleChange = jest.fn();
+		const { container, getByRole, getByTestId, getByLabelText } = render(
+			<SimpleSwitch label="A Switch" onChange={handleChange} />
+		);
+		const switchText = getByLabelText('A Switch');
+		const switchInput = getByRole('checkbox');
+		const switchToggle = container.querySelector('span[class$=-switch-toggle]') as HTMLElement;
+		const switchParent = getByTestId('switch-parent');
+		const user = userEvent.setup();
+		expect(switchInput).toHaveProperty('checked');
+		expect(switchInput).not.toBeChecked();
+		expect(switchInput).toHaveProperty('checked', false);
+		await user.click(switchText);
+		expect(switchInput).toBeChecked();
+		expect(switchInput).toHaveProperty('checked', true);
+		expect(handleChange).toHaveBeenCalledTimes(1);
+		await user.click(switchInput);
+		expect(switchInput).not.toBeChecked();
+		expect(switchInput).toHaveProperty('checked', false);
+		expect(handleChange).toHaveBeenCalledTimes(2);
+		await user.click(switchToggle);
+		expect(switchInput).toBeChecked();
+		expect(switchInput).toHaveProperty('checked', true);
+		expect(handleChange).toHaveBeenCalledTimes(3);
+		await user.click(switchParent);
+		expect(switchInput).not.toBeChecked();
+		expect(switchInput).toHaveProperty('checked', false);
```

Comment:
```suggestion
```
as above comments

Reply:
Thanks Tom, I have removed those in my latest commit pushed now. There's one thing that I might revert in the commit; that is a change to proptypes in Compactor.tsx and Repeater.tsx, due to running "Compactor.tsx and Repeater.tsx". Soobin knows and will take a look after lunch.